File: app/src/main/java/org/openweatherapp/viewmodels/CityViewModel.java
package org.openweatherapp.viewmodels;

import androidx.lifecycle.ViewModel;

import org.openweatherapp.models.City;

import java.util.ArrayList;
import java.util.List;

public class CityViewModel extends ViewModel {

    private List<City> mCities;

    public CityViewModel() {
        mCities = new ArrayList<>();
        mCities.add(new City("Abidjan"));
        mCities.add(new City("Lyon"));
        mCities.add(new City("Londres"));
    }

    public List<City> getCities() { return mCities; }
}

File: app/src/main/java/org/openweatherapp/ui/cities/CityViewModel.kt
package org.openweatherapp.ui.cities

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class CityViewModel : ViewModel() {

    val cities = listOf(
        City("Abidjan"),
        City("Lyon"),
        City("Londres")
    )

    private val _navigateToCityDetail = MutableLiveData<String>()
    val navigateToCityDetail: LiveData<String> = _navigateToCityDetail

    fun onCityClicked(name: String) {
        _navigateToCityDetail.value = name
    }

    fun onCityDetailNavigated() {
        _navigateToCityDetail.value = null
    }
}